packages/legend-art/src/panel/Panel.tsx

```diff
@@ -175,6 +175,107 @@ export const PanelFormSection = generateSimpleDIVComponent(
   'panel__content__form__section',
 );
 
+export const PanelFormValidatedTextField = forwardRef<
+  HTMLInputElement,
+  {
+    name?: string;
+    value: string | undefined;
+    update: (value: string | undefined) => void;
+    validateInput: (input: string) => string | undefined;
+    onValidationFailure?:
+      | ((validationIssue: string) => void | string | undefined)
+      | undefined;
+    prompt?: string | React.ReactNode;
+    placeholder?: string;
+    isReadOnly?: boolean;
+    className?: string | undefined;
+    errorMessageClassname?: string | undefined;
+    inputType?: string | undefined;
+    darkMode?: boolean;
+    fullWidth?: boolean;
+  }
+>(function PanelFormValidatedTextField(props, ref) {
+  const {
+    name,
+    value,
+    update,
+    prompt,
+    errorMessageClassname,
+    placeholder,
+    validateInput,
+    onValidationFailure,
+    isReadOnly,
+    className,
+    darkMode,
+    fullWidth,
+    inputType,
+  } = props;
+
+  const displayValue = value ?? '';
+
+  const [inputValue, setInputValue] = useState(displayValue);
+
+  const changeValue: React.ChangeEventHandler<HTMLInputElement> = (event) => {
```

Comment:
Just a comment really, I'm not sure yet what's the best way to handle it, but with this, when the value is changed elsewhere, we won't be able to update the form value (so is the problem of using `useState` in general)